.github/actions/setup-sentry/action.yml

```diff
@@ -0,0 +1,125 @@
+name: 'Sentry Setup'
+description: 'Sets up a Sentry test environment'
+inputs:
+  python:
+    description: 'python version (2 or 3)'
+    required: false
+    default: '3'
+  snuba:
+    description: 'Is snuba required?'
+    required: false
+    default: '1'
+  kafka:
+    description: 'Is kafka required?'
+    required: false
+    default: '0'
+
+outputs:
+  yarn-cache-dir:
+    description: "Path to yarn cache"
+    value: ${{ steps.config.outputs.yarn-cache-dir }}
+  pip-cache-dir:
+    description: "Path to pip cache"
+    value: ${{ steps.pip-cache.outputs.pip-cache }}
+  acceptance-dir:
+    description: "Path to acceptance visual snapshot artifacts"
+    value: ${{ steps.config.outputs.acceptance-dir }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup default environment variables
+      shell: bash
+      env:
+        PYTHON_VERSION: ${{ inputs.python }}
+        NEED_KAFKA: ${{ inputs.kafka }}
+      run: |
+        echo "PIP_DISABLE_PIP_VERSION_CHECK=on" >> $GITHUB_ENV
+        echo "MIGRATIONS_TEST_MIGRATE=0" >> $GITHUB_ENV
+        echo "SENTRY_LIGHT_BUILD=1" >> $GITHUB_ENV
+        echo "SENTRY_SKIP_BACKEND_VALIDATION=1" >> $GITHUB_ENV
+
+        [ "$PYTHON_VERSION" = "3" ] && echo "SENTRY_PYTHON3=1" >> $GITHUB_ENV || true
+
+        echo "NODE_ENV=development" >> $GITHUB_ENV
+        echo "NODE_OPTIONS=--max-old-space-size=4096" >> $GITHUB_ENV
+
+        # pytest-sentry configuration
+        echo "PYTEST_SENTRY_DSN=https://6fd5cfea2d4d46b182ad214ac7810508@sentry.io/2423079" >> $GITHUB_ENV
+        echo "PYTEST_ADDOPTS=--reruns 5" >> $GITHUB_ENV
+        [ "$GITHUB_REF" = "refs/heads/master" ] && echo "PYTEST_SENTRY_ALWAYS_REPORT=1" >> $GITHUB_ENV || true
+
+        # services configuration
+        # Note: some backend tests (e.g. tests/sentry/eventstream/kafka/test_consumer.py) will behave
```

Comment:
Can we set `USE_SNUBA` if `NEED_SNUBA`? It's already the case.

Reply:
Soooo this is a bit confusing -- for testing, we only use `USE_SNUBA` for the snuba test suite as there are some failing acceptance tests with snuba turned on. I suppose we should get this fixed so that we run with snuba by default.

Reply:
Added a comment below about this.